app/uk/gov/hmrc/agentclientrelationships/connectors/UsersGroupsSearchConnector.scala

```diff
@@ -51,4 +57,18 @@ class UsersGroupsSearchConnector @Inject()(
       case _: NotFoundException => Future failed RelationshipNotFound("UNKNOWN_AGENT_CODE")
     }
   }
+
+  def isAdmin(userId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Boolean] = {
+    val url = new URL(baseUrl, s"users-groups-search/users/$userId")
+    monitor("ConsumedAPI-UGS-getUserInfo-GET") {
+      httpGet.GET[CredentialRole](url.toString).map(_.credentialRole == "Admin")
+    } recoverWith {
+      case _: NotFoundException => Future failed UserNotFound("UNKNOWN_USER_ID")
+    }
+  }
+
+  def getAdminUserId(userIds: Seq[String])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[String] =
+    for {
+      admins <- Future.sequence(userIds.map(userId => isAdmin(userId).map(isAdminUser => (userId, isAdminUser))))
+    } yield admins.filter(_._2).map(_._1).headOption.getOrElse(throw new NoSuchElementException("no admin user"))
```

Comment:
ideally return `headOption` so the return type will be `Future[Option[String]]` instead of throwing exceptions everywhere  but that will be a big change

Reply:
I think that there has to be at least one admin in the list of userIds or they won't be authenticated to perform the next step, so I think it is an exceptional case

Reply:
yes, but once we have `Future[Option[String]]` we only proceed if this Option is defined else it will be a bad_request ?  currently it throws a server error,  is it ?